Hippo.Web/ClientApp/src/components/Order/CreateOrder.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from "react";
+import { OrderModel } from "../../types";
+import { useParams } from "react-router-dom";
+import { usePermissions } from "../../Shared/usePermissions";
+import { usePromiseNotification } from "../../util/Notifications";
+import OrderForm from "./OrderForm";
+import { authenticatedFetch, parseBadRequest } from "../../util/api";
+
+const defaultOrder: OrderModel = {
+  id: 0,
+  PILookup: "",
+  category: "",
+  name: "",
+  productName: "",
+  description: "",
+  notes: "",
+  units: "",
+  unitPrice: "",
+  installments: 5,
+  installmentType: "Yearly",
+  quantity: 0,
+  adjustment: 0,
+  adjustmentReason: "",
+  status: "N/A",
+  createdOn: "",
+  externalReference: "",
+  adminNotes: "",
+  subTotal: "",
+  total: "",
+  balanceRemaining: "",
+  metaData: [],
+  billings: [],
+  payments: [],
+  history: [],
+  percentTotal: 0,
+};
+const CreateOrder: React.FC = () => {
+  const { cluster, productId } = useParams();
+  const { isClusterAdminForCluster } = usePermissions();
+  const [order, setOrder] = useState<OrderModel>(null);
+  const [isClusterAdmin, setIsClusterAdmin] = useState(null);
+  const [notification, setNotification] = usePromiseNotification();
+
+  useEffect(() => {
+    setIsClusterAdmin(isClusterAdminForCluster());
+  }, [isClusterAdmin, isClusterAdminForCluster]);
+
+  useEffect(() => {
+    if (productId) {
+      const fetchProductOrder = async () => {
+        const response = await authenticatedFetch(
+          `/api/${cluster}/order/GetProduct/${productId}`,
+        );
+
+        if (response.ok) {
+          const data = await response.json();
+          setOrder(data);
+          // const balanceRemaining = parseFloat(data.balanceRemaining);
+          // setBalanceRemaining(balanceRemaining);
+          // const balancePending = data?.payments
+          //   .filter((payment) => payment.status !== "Completed")
+          //   .reduce((acc, payment) => acc + parseFloat(payment.amount), 0);
+          // setBalancePending(balancePending);
+        } else {
+          alert("Error fetching product for order");
+        }
+      };
+
+      fetchProductOrder();
+    } else {
+      if (isClusterAdmin === false) {
+        window.location.href = `/${cluster}/product/index`;
+      } else {
+        setOrder(defaultOrder);
+      }
+    }
+  }, [cluster, isClusterAdmin, productId]);
+
+  // async function so the form can manage the loading state
+  const submitOrder = async (updatedOrder: OrderModel) => {
+    const editedOrder: OrderModel = {
```

Comment:
maybe this is just for testing, but it seems like you shouldn't need any of it, just `use the updatedOrder` passed in.

Reply:
@laholstege ☝️  ? I think this was code you created, then I just added to it. (Or maybe I copied it from a different page...)